frontend/lib/resolution/resolution-queries.cpp

```diff
@@ -859,6 +861,25 @@ static Type::Genericity getFieldsGenericity(Context* context,
       }
     }
     return combined;
+  } else if (auto dt = ct->toDomainType()) {
+    Type::Genericity combined = Type::CONCRETE;
+
+    // Allows for instantiation of things like 'arg: domain'
+    // TODO: currently partially generic domains are not supported
+    if (dt->kind() == DomainType::Kind::Unknown) {
+      combined = Type::GENERIC;
+    }
+
+    return combined;
+  } else if (auto at = ct->toArrayType()) {
+    auto dt = getTypeGenericityIgnoring(context, at->domainType(), ignore);
+    auto et = getTypeGenericityIgnoring(context, at->eltType(), ignore);
+
+    if (dt != Type::CONCRETE || et != Type::CONCRETE) {
+      return Type::GENERIC;
+    } else {
+      return Type::CONCRETE;
+    }
```

Comment:
Should we try propagate a more specific genericity (concrete vs generic vs maybe generic?)
```suggestion
    if (dt == Type::GENERIC || et == Type::GENERIC) {
      return Type::GENERIC;
    } else if (dt == Type::MAYBE_GENERIC || et == Type::MAYBE_GENERIC) {
      return Type::MAYBE_GENERIC;
    } else {
       // or maybe another case for generic with defaults?
      return Type::CONCRETE;
    }
```

Reply:
Looking at the rest of this function, I think the general idea is to return either ``GENERIC`` or ``CONCRETE`` here.

``MAYBE_GENERIC`` doesn't seem like a useful piece of information to return from this function where arrays are concerned. If the domain or element type aren't concrete, then we know for sure it's generic. And arrays don't have "generic with defaults" behavior.